client/client_pool/src/concord_client_pool.cpp

```diff
@@ -221,8 +237,10 @@ SubmitResult ConcordClientPool::SendRequest(const bft::client::ReadConfig &confi
     callback(bftEngine::SendResult{static_cast<uint32_t>(OperationResult::INVALID_REQUEST)});
     return SubmitResult::Overloaded;
   }
+  auto request_flag = ClientMsgFlag::READ_ONLY_REQ;
+  if (config.request.reconfiguration) request_flag = ClientMsgFlag::RECONFIG_READ_ONLY_REQ;
   return SendRequest(std::forward<std::vector<uint8_t>>(request),
-                     ClientMsgFlag::READ_ONLY_REQ,
+                     request_flag,
```

Comment:
Simplify: 
auto request_flag = config.request.reconfiguration ? ClientMsgFlag::RECONFIG_READ_ONLY_REQ : ClientMsgFlag::READ_ONLY_REQ;

Reply:
Thanks @yuliasherman , I will add this in next commit.